### PR TITLE
[jenkins]: Add sonic-mgmt-framework build job

### PR DESCRIPTION
**Pull request by rupesh-k**
Add sonic-mgmt-framework build job

### CODE REVIEW

File: jenkins/common/sonic-mgmt-framework-build-pr/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    stages {
+        stage('Prepare') {
+            steps {
+                dir('sonic-mgmt-framework') {
+                    checkout([$class: 'GitSCM',
+                          branches: [[name: '${sha1}']],
+                          userRemoteConfigs: [[url: 'http://github.com/Azure/sonic-mgmt-framework',
+                                               refspec: '+refs/pull/*:refs/remotes/origin/pr/*']]])
+                }
+                copyArtifacts(projectName: 'vs/buildimage-vs-all', filter: '**/*.deb', target: 'buildimage', flatten: false)
```
Comment (qiluo-msft) [inline jenkins/common/sonic-mgmt-framework-build-pr/Jenkinsfile]: >vs [](start = 44, length = 2)

Do you mean `../vs` ?

File: jenkins/common/sonic-mgmt-framework-build-pr/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
+pipeline {
+    agent { node { label 'sonic-slave' } }
+
+    stages {
```
Comment (qiluo-msft) [inline jenkins/common/sonic-mgmt-framework-build-pr/Jenkinsfile]: Also add the option numToKeepStr ?

File: jenkins/common/sonic-mgmt-framework-build-pr/Jenkinsfile (hunk 1 above)
Comment (rupesh-k) [reply jenkins/common/sonic-mgmt-framework-build-pr/Jenkinsfile]: I am not sure how the path is picking up. Can you please help clarify.

If u see below pr jobs they have same path config as above. 

**lvnvde1934#cat sonic-swss-common-build-pr/Jenkinsfile | grep copyArt
                copyArtifacts(projectName: 'vs/buildimage-vs-all', filter: '**/*.deb', target: 'buildimage', flatten: false)

lvnvde1934#cat sonic-telemetry-build-pr/Jenkinsfile | grep copyAr
                copyArtifacts(projectName: 'vs/buildimage-vs-all', filter: '**/*.deb', target: 'buildimage', flatten: false)**


Only build jobs have ../vs path. 

lvnvde1934#cat sonic-swss-common-build/Jenkinsfile | grep copyArt
                copyArtifacts(projectName: '../vs/buildimage-vs-all', filter: '**/*.deb', target: 'buildimage', flatten: false)

File: jenkins/common/sonic-mgmt-framework-build-pr/Jenkinsfile (hunk 2 above)
Comment (rupesh-k) [reply jenkins/common/sonic-mgmt-framework-build-pr/Jenkinsfile]: None of pr jobs have numToKeepStr.  Is it some default value of 5 picked up for these jobs ?

lvnvde1934#pwd
/sonic-build-tools/jenkins/common

**lvnvde1934#grep -r numToKeep *pr
lvnvde1934#**

lvnvde1934#grep -r numToKeep *
dep-build/Jenkinsfile:        buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '10'))
linux-kernel-build/Jenkinsfile:        buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '10'))
python-asyncsnmp-build/Jenkinsfile:        buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '10'))
sonic-mgmt-framework-build/Jenkinsfile:        buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '10'))
sonic-swss-common-build/Jenkinsfile:        buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '10'))
sonic-swss-common-build-ubuntu/Jenkinsfile:        buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '10'))
sonic-telemetry-build/Jenkinsfile:        buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '10'))

**Comment by rupesh-k** [comment]
> As comments
I can make the changes as requested, but they are not matching the PR builds which are already present. Please let me know and I can make changes accordingly. 

Thanks

**Pull request merged**